### DISCUSSION TITLE
Permissions problems on root user

### DISCUSSION THREAD

**Discussion by kropsi** [directus/directus#13666] [Archive]
### Describe the Bug

I installed a fresh instance of Directus on the App Platform with a root admin. There was nothing special about this installation, but when creating collections and adding fields to them. I experience  Directus telling me that I lack the necessary permissions, even though I am a root admin user. If I repeat the task, then I succeed or fail again randomly. Sometimes it works, sometimes it doesn't.

### To Reproduce

Login with root admin and try to create collections and add fields to them.

### Errors Shown

It is quite sporadic how it fails but it always tells something with already existing fields or missing permissions.

Here is a short presentation of my experience: https://share.cleanshot.com/sOtngT

### What version of Directus are you using?

9.11.1

### What version of Node.js are you using?

16.15.0

### What database are you using?

Postgres 14 with PostGIS extension

### What browser are you using?

Not relevant

### How are you deploying Directus?

App Platform (Digital Ocean)

**Comment by azrikahar** [comment]
Granted we can never be absolutely sure without a reliable repro, but the "sporadic" nature does remind me of the possibility that you are being load-balanced between multiple instances, with one having the updated schema vs the other that doesn't have the new/removed collections/fields. (Note that although the error is 403, it is also possible that it's just missing. Ideally it shows 404 to lessen the confusion, but this is an existing topic for discussion in #4368)

With that said, are you running multiple instances of Directus with App Platform? Even if you aren't, would you mind sharing the [cache related env variables](https://docs.directus.io/configuration/config-options/#cache) that you may be using?

It would also be very helpful if you can verify whether this happens on local installations of Directus (without using pm2 as the cluster mode may induce the same behavior you are facing, provided there isn't a dedicated Redis instance in use)

**Comment by kropsi** [comment]
@azrikahar thank you for responding. I don't experience any problems in Docker locally. The only difference I made this time in App Platform is setting some env variables dynamically using `CONFIG_PATH=/workspace/config.live.yaml`.

I have one database and no load-ballancing. I had cache variables but I actually disabled them, as I thought that they gave me a problem. 
```
#CACHE_ENABLED: true
#CACHE_TTL: 30m
#CACHE_NAMESPACE: directus-cache
#CACHE_STORE: memory
#CACHE_AUTO_PURGE: true
#ASSETS_CACHE_TTL: 30m
```
I run a single instance of Directus on this database.

> It would also be very helpful if you can verify whether this happens on local installations of Directus (without using pm2 as the cluster mode may induce the same behavior you are facing, provided there isn't a dedicated Redis instance in use)

What do you mean by using pm2?

**Comment by azrikahar** [comment]
>  I don't experience any problems in Docker locally. 

Then I do believe (somehow) the only likely possibility is cache related. You did mention no load-balancing, but would you mind confirming the app is not scaled to more than 1 container? Reference: https://docs.digitalocean.com/products/app-platform/how-to/scale-app/

>  I had cache variables but I actually disabled them

`CACHE_SCHEMA` and `CACHE_PERMISSIONS` are actually independent of the `CACHE_ENABLED` env variable, and both of them does default to true. Perhaps you can try setting both of them to `false` and see how it goes?

Alternatively, you may also try clearing the cache manually first before trying the above solution. Reference on manual cache clear: https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache

> What do you mean by using pm2?

Some users do use pm2 cluster mode to run multiple instances of Directus, so if these instances are not using a shared Redis store, the behavior you have describe does happen here. The same issue can also appear when users run Directus on Kubernetes with multiple replicas (and this has been reported in the past), which is why the current suspect is on such interaction, since the "sporadic" nature really shouldn't happen on a single instance provided there are nothing else that we are not aware of here.
